### DISCUSSION TITLE
Issue storing multipule copies of one table in many other tables

### DISCUSSION THREAD

**Discussion by ronaldcohen1611** [prisma/prisma#20174] [Q&A] [answered]
Hello community, 

I am trying to build a simple News App, but I ran into this dilemma when working with prisma,

I have a basic schema where I have a table that stores "Articles" and another table called "customList" that can hold multiple articles. The problem I am facing is I can only store one reference of each row Articles inside one CustomList at a time. 

For example, lets say I have articles with ids [1, 2, 3, 4, 5] and I have two CustomList, 

A and B respectively. Lets say I add articles 1 & 2 to the list A and then try to add 2 & 3 to list B, the end result is list A having only article id 1 and list B has article ids 2 & 3. I am trying to see if there is anyway that I can get the articles to be stored uniqley in each custom list. At the moment I can only store one article to one list, but I am trying to store one article to multipole customList. If anyone could help me out with this bug I would greatly appreciate it. !! thanks in advance.

Here is my schema.prisma file for more context

```prisma
model Article {
  id           Int         @id @default(autoincrement())
  sourceId     String?
  sourceName   String?
  author       String?
  title        String
  description  String?
  url          String
  urlToImage   String?
  publishedAt  DateTime
  content      String?
  categoryId   Int?
  Category     Category?   @relation(fields: [categoryId], references: [id])
  customList   CustomList? @relation(fields: [customListId], references: [id])
  customListId Int?
}

model CustomList {
  id       Int       @id @default(autoincrement())
  name     String    @unique
  userId   Int
  user     User      @relation(fields: [userId], references: [id])
  articles Article[]
}

```

**Comment by ludralph** [comment, marked as answer]
Hello @ronaldcohen1611 👋 

Based on your schema, you have a one-to-many relationship between the `CustomList` and `Article` models, where each `CustomList` can have multiple associated `Article` records. 

You can modify the schema to define an explicit many-to-many relationship between CustomList and Article. This will involve introducing a new model, let's call it CustomListArticle, to represent the relation table.

```prisma
model Article {
  id          Int                 @id @default(autoincrement())
  sourceId    String?
  sourceName  String?
  author      String?
  title       String
  description String?
  url         String
  urlToImage  String?
  publishedAt DateTime
  content     String?
  categories  Category[] 
  customLists CustomListArticle[]
}

model CustomList {
  id       Int                 @id @default(autoincrement())
  name     String              @unique
  userId   Int
  user     User                @relation(fields: [userId], references: [id])
  articles CustomListArticle[]
}

model CustomListArticle {
  id           Int        @id @default(autoincrement())
  customListId Int
  articleId    Int
  customList   CustomList @relation(fields: [customListId], references: [id])
  article      Article    @relation(fields: [articleId], references: [id])

  @@unique([customListId, articleId])
}
```
In this updated schema, the `CustomListArticle` model represents the relation table between `CustomList` and `Article`. It has foreign key fields `customListId` and `articleId` that reference the primary keys of the respective models.

By using this explicit many-to-many relation, you can associate multiple articles with each custom list without overwriting the previous associations. The `CustomListArticle` model enables the storage of unique combinations of customListId and articleId using the @@unique attribute.

Once you've updated the schema, you can create, retrieve, and manage the many-to-many relationship between `CustomList` and `Article` through the `CustomListArticle` model.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ronaldcohen1611** [reply]
Hey @ludralph, thank you so much for this solution. I was able to get my feature working now. I really appreciate it !!

**Reply by ludralph** [reply]
I'm glad I could help 😄
